sdk/identity/identity/src/credentials/managedIdentityCredential/imdsMsi.ts

```diff
@@ -15,12 +15,15 @@ import { credentialLogger } from "../../util/logging";
 import { createSpan } from "../../util/tracing";
 import { imdsApiVersion, imdsEndpointPath, imdsHost } from "./constants";
 import { MSI, MSIConfiguration } from "./models";
-import { mapScopesToResource, msiGenericGetToken } from "./utils";
+import { mapScopesToResource } from "./utils";
 import { AuthenticationError } from "../../errors";
 
 const msiName = "ManagedIdentityCredential - IMDS";
 const logger = credentialLogger(msiName);
 
+/**
+ * Formats the expiration date of the received token into the number of milliseconds between that date and midnight, January 1, 1970.
+ */
 function expiresInParser(requestBody: any): number {
```

Comment:
Why is this the only copy of this function where we validate the existence of the `expires_on` property? Also, can we give `requestBody` an accurate type here?

Reply:
For sure! I’ll give requestBody a type! Thank you for pointing me to this. By this point it was invisible to me that there was an “any” type here.

Reply:
> Why is this the only copy of this function where we validate the existence of the expires_on property?

We assume that the other authentication environments will always provide an `expires_on` property.

In the cases where we don’t have a specific method, we always use `expries_in`, which should always be present.